src/content-tags-drawer/ContentTagsDropDownSelector.jsx

```diff
@@ -14,50 +15,42 @@ import './ContentTagsDropDownSelector.scss';
 import { useTaxonomyTagsData } from './data/apiHooks';
 
 const ContentTagsDropDownSelector = ({
-  taxonomyId, level, subTagsUrl, lineage, tagsTree, searchTerm,
+  taxonomyId, level, lineage, tagsTree, searchTerm,
 }) => {
   const intl = useIntl();
 
   // This object represents the states of the dropdowns on this level
   // The keys represent the index of the dropdown with
   // the value true (open) false (closed)
   const [dropdownStates, setDropdownStates] = useState({});
-  const isOpen = (i) => dropdownStates[i];
+  const isOpen = (tagValue) => dropdownStates[tagValue];
   const closeAllDropdowns = () => {
     const updatedStates = { ...dropdownStates };
     // eslint-disable-next-line no-return-assign
     Object.keys(updatedStates).map((key) => updatedStates[key] = false);
     setDropdownStates(updatedStates);
   };
 
-  const [tags, setTags] = useState([]);
-  const [nextPage, setNextPage] = useState(null);
+  const [numPages, setNumPages] = useState(1);
+  const parentTagValue = lineage.length ? decodeURIComponent(lineage[lineage.length - 1]) : null;
+  const {hasMorePages, tagPages} = useTaxonomyTagsData(taxonomyId, parentTagValue, numPages, searchTerm);
 
-  const [currentPage, setCurrentPage] = useState(1);
   const [prevSearchTerm, setPrevSearchTerm] = useState(searchTerm);
 
   // Reset the page and tags state when search term changes
   // and store search term to compare
   if (prevSearchTerm !== searchTerm) {
     setPrevSearchTerm(searchTerm);
-    setCurrentPage(1);
+    setNumPages(1);
     closeAllDropdowns();
```

Comment:
^ After thinking about this, now that I refactored the dropdowns to remember which tags are expanded or not by ID rather than page number, I think you should preserve the state even when the searchTerm changes, so just get rid of `closeAllDropdowns`. It will be much more convenient/consistent for users if the tags that they've opened stay open even if the search term changes.